stock_picking_location_check/models/stock_picking.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    allow_location_discrepancy = fields.Boolean(
+        help="If enable, the location discrepancies will be allowed between picking "
+        "and stock move lines."
+    )
+
+    @api.constrains("state")
```

Comment:
Shouldn't we use `location_id`, `location_dest_id`, `move_line_ids.location_id`, and `move_line_ids.location_dest_id`, instead?

Reply:
Unless I misunderstood about the behavior of `@api.constrains` decorator, it is only triggered when these fields are included in the create and write call. So, when we change location_dest_id or location_id of move lines from detailed operations, these function will not trigger.